### PR TITLE
doc: Update init and reduce-traffic docs for -blocksonly

### LINKED ISSUE #17294: code for blocksonly disagrees with documentation
The docs for `blocksonly` say that "Transactions from the wallet or RPC are not affected" but the code says:

    if (gArgs.GetBoolArg("-blocksonly", DEFAULT_BLOCKSONLY) && gArgs.SoftSetBoolArg("-walletbroadcast", false)) {
        LogPrintf("%s: parameter interaction: -blocksonly=1 -> setting -walletbroadcast=0\n", __func__);
    }

Here's the relevant documentation:

    -blocksonly
       Whether to reject transactions from network peers. Transactions from the
       wallet or RPC are not affected. (default: 0)

    -walletbroadcast
       Make the wallet broadcast transactions (default: 1)

Doesn't that imply that `blocksonly` *does* affect transactions from the wallet?

### PR DESCRIPTION
**Pull request by glowang**
When -blocksonly is set to 1, it interacts with the -walletbroadcast
parameter and sets it to 0. 

This behavior is not captured by the current documentation, which
claims that -blocksonly does not impact any wallet transactions at
all.

Fixes #17294

### CODE REVIEW

**Review by maflcko — COMMENTED** [review]
* The walletbroadcast is only set "softly" to `0`, i.e. when it was not set otherwise (in the config or on the command line)
* RPC transactions *are* relayed as far as I can tell, see also the second section in the test `./test/functional/p2p_blocksonly.py`
* It would be nice to extend that test for wallet transactions and maybe even whitelisted peers. (Not necessarily in this pull, it can also be done later)

**Comment by andrewtoth** [comment]
Lines `6` and `29` of `reduce-traffic.md` still show the outdated number of `8` outgoing peers, but this has been changed to `10`. Since you're modifying that file you could update those numbers as well.

**Comment by glowang** [comment]
> * The walletbroadcast is only set "softly" to `0`, i.e. when it was not set otherwise (in the config or on the command line)
> * RPC transactions _are_ relayed as far as I can tell, see also the second section in the test `./test/functional/p2p_blocksonly.py`
> * It would be nice to extend that test for wallet transactions and maybe even whitelisted peers. (Not necessarily in this pull, it can also be done later)

Are you referring to “wallet transaction" as a loose term? I thought wallet transaction includes the ones initiated via rpc already?

File: doc/reduce-traffic.md [hunk 1]
```diff
@@ -48,3 +48,7 @@ Be reminded of the effects of this setting.
   wallet is loaded or if you use the node to broadcast transactions.
 - It makes block propagation slower because compact block relay can only be
   used when transaction relay is enabled.
+- This setting sets the flag "-walletbroadcast" to be "0", if it is currently
+  set to "1", which disables the automatic broadcasting of transactions
```
Comment (MarcoFalke) [inline doc/reduce-traffic.md]:
```suggestion
  unset, which disables the automatic broadcasting of transactions
```

The argument is set "softly", see also https://doxygen.bitcoincore.org/class_args_manager.html#ae77153300424d36e02e54f2c718a457f

File: doc/reduce-traffic.md [hunk 2]
```diff
@@ -48,3 +48,7 @@ Be reminded of the effects of this setting.
...
+  from inbound peers. If a peer is whitelisted, and "-whitelistforcerelay" is
```
Comment (MarcoFalke) [inline doc/reduce-traffic.md]:
```suggestion
  from the wallet. If a peer is whitelisted, and "-whitelistforcerelay" is
```

File: doc/reduce-traffic.md [hunk 3]
```diff
@@ -48,3 +48,7 @@ Be reminded of the effects of this setting.
   wallet is loaded or if you use the node to broadcast transactions.
 - It makes block propagation slower because compact block relay can only be
   used when transaction relay is enabled.
+- This setting sets the flag "-walletbroadcast" to be "0", if it is currently
```
Comment (MarcoFalke) [inline doc/reduce-traffic.md]: It could probably make sense to move up this bullet point to after the second bullet point, since they are related.

File: doc/reduce-traffic.md [hunk 4]
```diff
@@ -26,7 +26,7 @@ calculating the target.
 
 ## 2. Disable "listening" (`-listen=0`)
 
-Disabling listening will result in fewer nodes connected (remember the maximum of 8
+Disabling listening will result in fewer nodes connected (remember the maximum of 10
```
Comment (andrewtoth) [inline doc/reduce-traffic.md]: Line `6` also has the outdated number.

Comment (laanwj) [reply doc/reduce-traffic.md]: Yup, might want to change the number of peers (and the composition: 8 full outgoing connections and two blocks-only).

File: src/init.cpp [hunk 5]
```diff
@@ -371,7 +371,7 @@ void SetupServerArgs()
     gArgs.AddArg("-blocknotify=<cmd>", "Execute command when the best block changes (%s in cmd is replaced by block hash)", ArgsManager::ALLOW_ANY, OptionsCategory::OPTIONS);
 #endif
     gArgs.AddArg("-blockreconstructionextratxn=<n>", strprintf("Extra transactions to keep in memory for compact block reconstructions (default: %u)", DEFAULT_BLOCK_RECONSTRUCTION_EXTRA_TXN), ArgsManager::ALLOW_ANY, OptionsCategory::OPTIONS);
-    gArgs.AddArg("-blocksonly", strprintf("Whether to reject transactions from network peers. Transactions from the wallet, RPC and relay whitelisted inbound peers are not affected. (default: %u)", DEFAULT_BLOCKSONLY), ArgsManager::ALLOW_ANY, OptionsCategory::OPTIONS);
+    gArgs.AddArg("-blocksonly", strprintf("Whether to reject transactions from network peers. Automatic broadcast and rebroadcast of any transactions from inbound peers is disabled, unless '-whitelistforcerelay' is '1', in which case whitelisted peers' transactions will be relayed. rpc transactions are not affected. (default: %u)", DEFAULT_BLOCKSONLY), ArgsManager::ALLOW_ANY, OptionsCategory::OPTIONS);
```
Comment (laanwj) [inline src/init.cpp]: please capitalize 'RPC'

**Comment by laanwj** [comment]
Thanks for improving the documentation!
ACK minus nits

File: doc/reduce-traffic.md (hunk 1 above)
Comment (glowang) [reply doc/reduce-traffic.md]: Ah got it. Thanks!

File: doc/reduce-traffic.md (hunk 2 above)
Comment (glowang) [reply doc/reduce-traffic.md]: sorry if this is a super naive question...my understanding is transactions from inbound peers and wallet transactions are two different concepts. So actually both of them not relayed when -blocksonly mode is turned on?

Comment (MarcoFalke) [reply doc/reduce-traffic.md]: Yes, but walletbroadcast is a wallet specific setting and does not control incoming traffic, only outgoing (broadcast)
I think it could help to split the walletbroadcast and whitelist settings in two bullet points.

File: doc/reduce-traffic.md (hunk 1 above)
Comment (glowang) [reply doc/reduce-traffic.md]: I still have a clarifying question about "softset". My understanding is that this param interaction is not guranteed to happen. It would only happen if the setting was not manually set already by the user. 

Therefore, If the user sets -walletbroadcast to 1 first, and later sets -blocksonly to 0, this would effectively means that 1) -walletbroadcast will not be set to 0 and 2) all transaction broadcast happens normally?

Comment (MarcoFalke) [reply doc/reduce-traffic.md]: > Therefore, If the user sets -walletbroadcast to 1 first, and later sets -blocksonly to 0, this would effectively means that 1) -walletbroadcast will not be set to 0 and 2) all transaction broadcast happens normally?

Yes.

File: doc/reduce-traffic.md (hunk 4 above)
Comment (glowang) [reply doc/reduce-traffic.md]: > Yup, might want to change the number of outgoing peers there (and mention the composition: 8 full outgoing connections and two blocks-only).

This would probably mean that the max connection now is 127, since we have 10 outgoing and 117 incoming?

Comment (MarcoFalke) [reply doc/reduce-traffic.md]: No, DEFAULT_MAX_PEER_CONNECTIONS is still 125

**Comment by laanwj** [comment]
ACK 869327ca6e3d5feccb477e4a04e9b894bc2cdddd

File: doc/reduce-memory.md [hunk 6]
```diff
@@ -24,8 +24,8 @@ The size of some in-memory caches can be reduced. As caches trade off memory usa
 
 ## Number of peers
 
-- `-maxconnections=<n>` - the maximum number of connections, this defaults to `125`. Each active connection takes up some memory. Only significant if incoming
-   connections are enabled, otherwise the number of connections will never be more than `8`.
+- `-maxconnections=<n>` - the maximum number of connections, this defaults to `127`. Each active connection takes up some memory. Only significant if incoming
```
Comment (MarcoFalke) [inline doc/reduce-memory.md]:
```
$ ./src/qt/bitcoin-qt -?|grep -C 2 'maxconn'
QSocketNotifier: Can only be used with threads started with QThread
       Automatically create Tor hidden service (default: 1)

  -maxconnections=<n>
       Maintain at most <n> connections to peers (default: 125)

File: src/net.h [hunk 7]
```diff
@@ -64,7 +64,7 @@ static const unsigned int MAX_SUBVERSION_LENGTH = 256;
 /** Maximum number of automatic outgoing nodes over which we'll relay everything (blocks, tx, addrs, etc) */
 static const int MAX_OUTBOUND_FULL_RELAY_CONNECTIONS = 8;
 /** Maximum number of addnode outgoing nodes */
-static const int MAX_ADDNODE_CONNECTIONS = 8;
+static const int MAX_ADDNODE_CONNECTIONS = 10;
```
Comment (MarcoFalke) [inline src/net.h]: Code changes (behaviour changes) should probably be in a separate pull request from documentation changes.

**Review by maflcko — APPROVED** [review]
ACK after the code change is removed

**Comment by MarcoFalke** [comment]
ACK 621e86ee8d0102e2bf41f7656a368083b89b2f83

File: doc/reduce-memory.md [hunk 8]
```diff
@@ -24,8 +24,7 @@ The size of some in-memory caches can be reduced. As caches trade off memory usa
 
 ## Number of peers
 
-- `-maxconnections=<n>` - the maximum number of connections, this defaults to `125`. Each active connection takes up some memory. Only significant if incoming
-   connections are enabled, otherwise the number of connections will never be more than `8`.
+- `-maxconnections=<n>` - the maximum number of connections, this defaults to `125`. Each active connection takes up some memory. Only significant if incoming connections are enabled, otherwise the number of connections will never be more than `10`. Of the 10 outbound peers, there can be 8 full outgoing connections and 2 -blocksonly peers, in which case they are block/addr peers, but not tx peers.
```
Comment (MarcoFalke) [inline doc/reduce-memory.md]: whoopsie, just realized that blocksonly is different from block-relay-only. Fixed in #18464
